addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/internal/X10Interface.java

```diff
@@ -0,0 +1,788 @@
+
+    /**
+     * Flag to indicate that background thread should be killed. Used to deactivate plugin.
+     */
+    protected volatile boolean killThread = false;
+
+    // Scheduling attributes
+    /**
+     * Queue of as-yet un-actioned requests.
+     */
+    protected BlockingQueue<Cm11aAbstractHandler> deviceUpdateQueue = new ArrayBlockingQueue<Cm11aAbstractHandler>(256);
+
+    // Need to keep last addresses found for data that comes in over the serial interface because if the incoming
+    // command is a dim or bright the address isn't included. In addition some controllers will send the address
+    // in one message and the function in a second one
+    private List<String> lastAddresses;
+
+    /**
+     *
+     * @param serialPort serial port device. e.g. /dev/ttyS0
+     * @throws NoSuchPortException
+     *
+     */
+    public X10Interface(String serialPort) throws NoSuchPortException {
+        super();
+        logger.trace("**** Constructing X10Interface for serial port: {} *******", serialPort);
+        portId = CommPortIdentifier.getPortIdentifier(serialPort);
+    }
+
+    /**
+     * Establishes a serial connection to the hardware, if one is not already established.
+     */
+    protected boolean connect() {
+        if (!connected) {
+            if (serialPort != null) {
+                logger.trace("Closing stale serialPort object before reconnecting");
+                serialPort.close();
+            }
+            logger.debug("Connecting to X10 hardware on serial port: {}", portId.getName());
+            try {
+                serialPort = portId.open("Openhab CM11A Binding", IO_PORT_OPEN_TIMEOUT);
+                serialPort.setSerialPortParams(4800, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                        SerialPort.PARITY_NONE);
+                serialPort.setFlowControlMode(SerialPort.FLOWCONTROL_NONE);
+                serialPort.disableReceiveTimeout();
+                serialPort.enableReceiveThreshold(1);
+
+                serialOutputStr = serialPort.getOutputStream();
+                serialOutput = new DataOutputStream(serialOutputStr);
+                serialInputStr = serialPort.getInputStream();
+                serialInput = new DataInputStream(serialInputStr);
+
+                serialPort.addEventListener(this);
+                connected = true;
+
+                serialPort.notifyOnDataAvailable(true);
+                serialPort.notifyOnRingIndicator(true);
+
+                // Bob Raker - add serial timeout to prevent possible hang conditions
+                serialPort.enableReceiveTimeout(SERIAL_TIMEOUT_MSEC);
+                if (!serialPort.isReceiveTimeoutEnabled()) {
+                    logger.info("Serial receive timeout not supported by this driver.");
+                }
+            } catch (PortInUseException e) {
+                logger.warn("Serial port {} is in use by another application ({})", portId.getName(), e.currentOwner);
```

Comment:
I did a quick check, but do you properly update the bridge connection state based on this exceptions. Please validate this and also make the bridge status 'always' correctly represents the real status.